### ISSUE TITLE
Option to disable auto updates

### ISSUE THREAD

**Issue by Golgo1** [a85/POSTMan-Chrome-Extension#6999]
**Describe the bug**
Postman updates even when Auto updates are turned off

**To Reproduce**
Run a slightly older version of Postman
Postman will update, breaking critical functionality

**Expected behavior**
Don't update until directed to

**Screenshots**
If applicable, add screenshots to help explain your problem. Please make sure the screenshot does not contain any sensitive information such as API keys or access tokens. 

**App information (please complete the following information):**
 - App Type:  Native App
 - Postman Version: 7.2.2 forced to 7.3.4
 - OS: Win 10

**Additional context**
Yes, this has been raised before and closed.  This should not be ignored.

**Comment by preethammavin** [comment]
@Golgo1 let us know what is the critical functionality so that we can get it sorted for you. Postman enables updates being turned off for major versions as we expect that it might break user workflows. 
Where as minor versions only include updates to current system to ensure we continuously improve the product without breaking user workflows by pushing incremental feature updates, bug fixes and security bumps. We treat regressions as an unacceptable class of bugs at Postman and we strive to ensure it does not happen, please do let us know what are functionalities that you feel are broken in the release and we shall ensure that it is fixed.

**Comment by Golgo1** [comment]
Thank you for the reply.  The primary bug relates to SSL connections.  I've created bug 7000 to address that.
I've noticed a few other GUI-related bugs as well.  In fact that's how I knew I was updated. (I'll make separate reports for those)
I was force updated a month ago to 7.3.3.. BEFORE it was released on the website.  The Latest was still 7.2.2.
7.3.3 broke many of the same features, but I was able to re-install 7.2.2 and work for a few weeks.
7.3.4 is force updating instantly.  

I appreciate the quick responses on both bug reports, but I need to spend my morning configuring a different tool so I can try and meet my project deadline
I know this is not the place for complaints, but I feel having the option to control their own updates is a feature the community desperately needs, and is being ignored

**Comment by UncleBravado** [comment]
I fully support that requests as many other users in comments to different tickets. I do not understand why to force update, especially if it is not about minor updates or patches (as you claim) - for instance with version 7.3.X there were significant updates with many regression bugs. I do not think that the complete change in component rendering request's response is a minor change. Postman is not a toy, it is a serious tool for many of us and we wasted a lot of time to revert the version, just to repeat this exercise with every restart.

Just imagine that it happens to your phone and it gets unusable every morning.

I appreciate your approach to regression bugs but it is just impossible to avoid them, so maybe better let the desperate users have the possibility to control the updates and do not ignore that request as your decision to force updates seems to be totally against Postman users.

**Comment by robross0606** [comment]
This is a huge problem with your design that needs to be addressed ASAP.  Updates need to be able to be shut off COMPLETELY.  Frankly, no one cares what you classify as a "minor" vs "major" upgrade.  People turn off updates because they need consistent repeatable behavior, want to limit network usage, and more.  Many people need a tool that's not going to change out from underneath them at any time, regardless of whether the developer thinks (often incorrectly) that this is a "minor" change.  This is also a red flag security risk for places where version control is critical and software is tracked.

**Comment by robross0606** [comment]
Case in point, Postman just updated AGAIN to 7.5.0 even though I have updates turned OFF.  Not only that, but it doesn't even give you a change log or any indication of what was actually updated.  This is unacceptably heavy-handed.

**Comment by robross0606** [comment]
Perhaps "major" updates should be considered anything that isn't a third level point release?  How is the move from 7.2 to 7.5 not considered a "major" release?  Look at the [change log](https://www.getpostman.com/downloads/release-notes)!  Just the update to 7.3 has a massive number of potentially breaking changes.  A whole OS architecture support was removed and a completely new (and not thoroughly tested) richtext editor was swapped in.  Massive fail.

**Comment by ZooDoo4U** [comment]
Would like to point out updates are a pain,  as one is not able to save custom setting for fonts and colors. Asking on the forums a person showed how to de-compile the resource file, after which I then fix up the css files to get the fonts and colors desired.   I'm not doing this to make it look nice, i'm doing it so I can simply use Postman.  Thought after setting "Disable automatic updates" would fix is.  Seems wrong for two reasons.  First it seems backwards,  minor updates should be just that, minor.  Secondly where is the prompt to allow a person to opt out if the update will be forced on them?  Finally there should be an option if there is a critical security issue, ah but that we can all agree a security fix would not be a minor issue...  But simply adding minor spacing or "minor" updates, should be completely optional if a user doesn't want the update.

**Comment by prdemelof** [comment]
I finally found a way to completely and totally disable the dreaded auto updates. Be it minor, major or whatever. Bit of a cheat though, but it gets the job done until the team fixes it. Navigate to "C:\Users\USERNAME\AppData\Local\Postman". In this folder you will find and delete everything that has "update" && ".exe" in it's name. While you are at it, notice the previous updates are all still there, filling up your SSD (assuming you run your OS on SSD). Yeah, go ahead and trash those too.

You're welcome.

![optional-updates](https://user-images.githubusercontent.com/3854058/96359000-1900e400-1140-11eb-8749-06d6d8d52811.png)

**Comment by stefanluchian** [comment]
@ Postman guys: Why are you insisting to force us to update, even on minor updates?
This is no way to win new users! And you might lose existing ones.
We simply don't want automatically minor updates! Is it so hard to understand it?
We are not some mobile-app-freaks who just want to have the latest thing.
Just let us decide if and when we want minor updates!
This is really annoying!
Stop forcing us to update! Should be our decision!

**Comment by arlemi** [comment]
@stefanluchian As mentioned by @preethammavin above, minor updates should not break your workflows, and are here to make sure you have the latest security updates and bug fixes. 

You can disable major updates in your app settings, and if you'd like to stay on v7 for now please reach out to help@postman.com and they'll help you revert to it. 

P.S: I've removed your comment from the other issues as this is was the same one copied/pasted and these issues were closed.

**Comment by jyeros** [comment]
@arlemi I agree that minor updates should not break user workflows and they provide users with the latest security updates and bug fixes, it is really useful. But this applies not only to the Postman app, it applies to every software we use, it doesn't care if it is a simple calculator or an entire operating system. Following your comment all software must force users to update, and it doesn't happen. Users should have the ability to decide (People over tools).

 I dislike when the updates are forced because they can have a significant impact over metered connection and minor updates should not break anything but we are humans we can make mistakes, it is because security and bug issues exist. You can discontinue the old version if you like, I'd rather it stop working than not being able to decide. 

If you implement this, users who use automatic updates will not be affected. i don't see a good reason to not implement it.

**Comment by robross0606** [comment]
The arrogance of these statements is that the word "should" keeps being thrown around.  The fact is, we're all human.  Humans make mistakes.  People should have the _right_ to stick with a particular version that they _know_ is stable and update on _their_ terms.  Not when you deem it necessary and then accidentally push a breaking bug.  We've already seen how this works out when you "fix" something and cause a bad regression bug on a minor update.  How many of those instances have to happen before you listen to your users and allow us to _STOP_ automatic updates, even for what you deem a "minor fix"?

**Comment by ZooDoo4U** [comment]
Agreed,  most programs "SHOULD" allow users the ability to work as they desire.  Programmers "SHOULD" remember they are the ones providing a service to what users need.   Seems like such an easy solution here?  "SHOULD" be a pretty simple question before updating...  "SHOULD"  ask "Would you like to update "Yes/No"?  The program "SHOULD" honor a users choice and back out.  

It seems like this "SHOULD" be a simple fix and all would be happy.  What are we missing?

**Comment by Stevemoretz** [comment]
I hope this doesn't come off rude.

But I **don't care** what you want for now, All I know is that I don't want my data to be wasted on something that I don't wanna have.So I either delete it and use an alternative or do a workaround.

Here's my workaround add these to your hosts file:

```
0.0.0.0         dl.pstmn.io
0.0.0.0         sync-v3.getpostman.com
0.0.0.0         getpostman.com
```

If you have a better one please **do** tell us.

**Comment by sojimaxi** [comment]
@arlemi Honestly speaking it is absolutely rude to force your ideas on your users. I should have the freedom to choose when I want to update. Your part is to notify us that there is an update available in whatever version (Major, Minor or Patch).

For instance when am out of the office I set my data to metered but your application doesn't respect that either. It happened again yesterday, it went ahead to just update without notification or options. I noticed the spike in data consumption and checked the task manager to see why.
Let us know an update is available `Update to version 7.x.x (100MB)` and a button for us to decide if we want to. 

Please **respect** the user's **choice of disable automatic updates**. Most people who disable that are aware why they wanted it that way.

**Comment by UncleBravado** [comment]
@arlemi I'm a bit speechless, frankly. You have quite a lot of reports, plenty of arguments from different users, dozens of samples how it turned wrong. From your side, I'm sorry guys, there is nothing but repeated "should".

I see no single **real** reason not to implement PROPERLY that bloody option in your app.
Do you really want us to do tricks with limiting your app connectivity? You know that we can easily do that, some or other way but it makes no sense to ruin your picture in our eyes.

If this is about security (I doubt), just wrap this setting with tons of warnings but allow it. Browsers can do for instance...

**Comment by ZooDoo4U** [comment]
@UncleBravado Have to agree with you about the "Security Issue".  Considering my use case with Postman, 99.99999% of the time I’m hitting my own end point for development.  Normally the only thing I want is json, that's it.   And json is or should be text and read only?  No reason to be executing anything with it.
 
If not giving a “Disable Updates” is a "security issue" first question is why is Postman even attempting to execute data from any response, or are there bigger problem in Postman why it is a security risk to use?  Shouldn't responses be display only anyway?  If not there should be an option to prevent all scripts from executing on response data?  Would assert for security, the setting be off by default and users would need to explicitly enable running scripts.  Safe to assert Postman's assumption to run scripts by default is wrong and that is a much bigger security hole.   Safe to assert most naive Postman users may have no clue scripts will be executed automatically.  Additionally, my experience the same naive users will tend to run with elevated permissions to simplify things, making the execute scripts in Postman by default even worse?  

Would seem the real issue, if not allowing a "Turn off updates"   really is a "Security hole", fix the real security hole in Postman by preventing scripts to execute when not desired.

**Comment by Stevemoretz** [comment]
Force update for security makes not that much sense.
What kind of security are we talking about here?
Even if this is about security that makes sense to happen on big updates not minor ones.


@ZooDoo4U @UncleBravado 
Here's a workaround to turn off your updates though which worked for me:
https://github.com/postmanlabs/postman-app-support/issues/6999#issuecomment-770186680

**Comment by prdemelof** [comment]
For Windows only :(

I'm feeling a bit guilty because I've been keeping this to myself but, here it goes: found another way to completely and totally disable the dreaded auto updates. Be it minor, major or whatever. Bit of a cheat though, but it gets the job done until the team fixes it. Navigate to "C:\Users\USERNAME\AppData\Local\Postman". In this folder you will find and delete everything that has "update" && ".exe" in its name. You might wanna keep backups, just in case you decide to update in future. While you are at it, notice the previous updates are all still there, filling up your SSD (assuming you run your OS on SSD). Yeah, go ahead and trash those too.

![image](https://user-images.githubusercontent.com/3854058/106586592-9e8b9100-6583-11eb-89e3-0bf9f2a46eb9.png)

You're welcome.

**Comment by Stevemoretz** [comment]
Your solution isn't cross platform.
But gives me an idea there is probably the same update executable inside /Application/Postman

Which via deleting it the same thing happens on macos.

Haven't tried it yet.
Hosts solution is working for me for now.

But opera browser has the same updater executable and deleting that disabled the updates.

**Comment by prdemelof** [comment]
@Stevemoretz Oh yeah you're right. I should have mentioned my solution was for Windows. 😅

**Comment by bwosborne2** [comment]
I am running postman 7.x and have no interest in Postman 8 which is a major update. 

I see no reasonable way to disable the notifications even though I do not want it. Auto download is turned off but it still notifies every time I run the app.

It appears Postman 8 requires a cloud connection to use workspaces, breaking the local workspace option in previous versions.

**Comment by Stevemoretz** [comment]
> I am running postman 7.x and have no interest in Postman 8 which is a major update.
> 
> I see no reasonable way to disable the notifications even though I do not want it. Auto download is turned off but it still notifies every time I run the app.
> 
> It appears Postman 8 requires a cloud connection to use workspaces, breaking the local workspace option in previous versions.

Switch back to v7, block connection to postman servers.
https://stackoverflow.com/a/65966655/10268067

**Comment by vbiginner** [comment]
Please do not force downloading of minor updates. I do work on limited bandwidth and data. 
Earlier was not aware of this and I noticed when all my mobile data wiped out at the start of the day.
Stop auto updating Postman.

**Comment by FikaCode** [comment]
so, It's been almost two years, Is there any official way to disable update？

**Comment by ZooDoo4U** [comment]
This may not be a fix, but found a useful work around.  Been starting to use VS Code, and found there iis a VS like plug-in that will do all what i need without postman and without issues.  Simple clean and avoids postman altogether.  Works for me.   Have a look for "Thunder Client",  even being new and running into an issue and reported it, the author very proactive, helpful and mentioned what I was asking for a new feature and even better i was in beta and getting pushed, and coming in the next update.  Fair enough but a HUGE difference from the impertinent responses  received from the postman people.  Guess i'm now a 'post' postman person.  Never thought we were asking for too much or that it was unreasonable, hard to see how no update means updates are still on...

**Comment by Y-Less** [comment]
Any update on this?  I disable automatic updates on principle alone.  I do not like the idea of someone else deciding when they can install software on my computer.  I just got an update today and thought I had forgotten to disable updates when I reinstalled a few weeks ago.  But no, you're just ignoring the setting to disable them, i.e. ignoring a setting I set on my computer to control what software running on my computer does.  If I set a setting, I expect you to respect it!

**Comment by Y-Less** [comment]
In the meantime, I tried the work-around I've used for Chrome:

* Go to `C:\Users\<username>\AppData\Local\Postman\`.
* Delete `Update.exe`.
* Create a new empty file and call it `Update.exe`.
* Set the permissions so no-one can edit it (`Deny` everyone).

I tested this by reinstalling 8.10.0, but the program instantly crashes on startup.

Edit:  Found a way to do it (actually much simpler):

* Just set `Deny` on the write permission for everyone on the whole of the `C:\Users\<username>\AppData\Local\Postman\` directory.

**Comment by ZooDoo4U** [comment]
I found a great work around and many can easily take advantage of it.   As more and more of us are being consumed by the VS Code Editor Virus  (hahaha)  there is an extension for VS Code called "Thunder Client" that will do most of what i need from Postman.  Even better the developer of the extension when reported issues is very responsive and even better very friendly and receptive to user's desires,  completely the opposite of my experience from Postman.   Would suggest you try the Thunder Client and see if it fits you needs.

**Comment by nitty8585** [comment]
We make Postman available to our customers through an MDM. End users do not have admin rights, so they are prompted to update, which then obviously fails. Why would you give an option to disable updates, then force them anyway? An update is an update - major or minor is irrelevant.

**Comment by Stevemoretz** [comment]
You can easily disable it, check out the updated workaround:

https://stackoverflow.com/a/65966655/10268067

**Comment by nitty8585** [comment]
We need something similar for macOS. @Stevemoretz do you know of any equivalent?

**Comment by Stevemoretz** [comment]
> We need something similar for macOS. @Stevemoretz do you know of any equivalent?

I'm using it on macos myself, it works on macos windows and Linux, please read the complete answer with details at https://stackoverflow.com/a/65966655/10268067

**Comment by arakis** [comment]
I'm working with a metered connection too, and the automatic minor updates are absolutly not acceptable. as @Stevemoretz  mentioned, you simply can block the download domains of postman, just add to your hosts file:

```
0.0.0.0         dl.pstmn.io
0.0.0.0         sync-v3.getpostman.com
0.0.0.0         getpostman.com
0.0.0.0         go.pstmn.io
```

As you see, this is not a nice way, and have disadvantages. So, please offer an option to allow disabling minor updates.

Just an usefull Tipp, that could be acceptable: Please check if the network connection has the "Metered" Flag. If Set, don't download. If not set (in 99.9% of the cases of most users)), just download like yet.

**Comment by CLannadZSY** [comment]
2021/12/02, still exists!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
why??????????????????????????????????????????????????????????????

**Comment by TheArKaID** [comment]
It's one year since the last Collaborator responding here. I just wondering why it's so hard to make it off permanently ? It should be as easy as toggling a switch button with true or false value, isn't it ? Why are you guys even recommending a workaround by editing the hosts file instead of adding a simple-small-switch-button in the Setting ?

**Comment by Stevemoretz** [comment]
> Why are you guys even recommending a workaround by editing the hosts file instead of adding a simple-small-switch-button in the Setting ?

Because we don't work here, we're just like you but found a way so we share it so other people can use it too.

**Comment by sylvix** [comment]
Ok, so there actually is a BIG breaking problem with auto-updates. I'm working with Postman on Linux, so when I run it to quickly check an endpoint and then shut it down after couple of minutes, it doesn't download the update completely (as it seems) and after that the whole app breaks - the Postman symlink doesn't work anymore and I have to unpack it again from the tar archive.

**Comment by jviotti** [comment]
@sylvix Would you mind opening a new ticket describing this behavior so that we can discuss it in more depth there?

**Comment by jviotti** [comment]
Hey @sylvix , as a follow up, we have confirmed that the bug you mentioned was already discussed and fixed in 9.14.0 over here: https://github.com/postmanlabs/postman-app-support/issues/10679. Let us know it that solves the problem for you!

**Comment by CalvinTG** [comment]
Hi,
we deploy Postman via Software distribution and our users don't have administrative rights.
Therefore users cannot update Postman themselves.
We switched off the major updates in the settings, but as many users mentioned in this blog we also cannot switch off minor updates.
Generally in a modern software we expect a switch to turn off ALL updates. This also coud be a hidden switch, available only for admins deploying software to multiple users.
Actually we don't want to block the dowload domains via the hosts file. 
Therefore it would be great, if Postman will move forward to integrate such a switch into the software to help users like us.

**Comment by sylvix** [comment]
@jviotti
Thanks for the fast reply! I'll check if it works. Right now Postman shows me version 9.14.0, so I'll see how that goes. 

Anyway I still think that a quick setting to turn it off would be more preferable to everyone.

**Comment by maclermo** [comment]
Agreed ! I would like to be able to skip updates altogether.

**Comment by ilias-sp** [comment]
i am rarely using Postman. Postman causes more network usage due to downloading updates that i cannot control, offering advanced features that i dont need, rather than the actual traffic i am generating triggering requests. 

Please change the app, Downloading major/minor/any kind of updates should be controlled by the end user.

**Comment by merlinstardust** [comment]
Is there an ETA when this behavior will be corrected?

This issue has been open for 3 years, so it's a bit baffling that this is still an open issue. Can someone explain why it's taking 3+ years to add a simple toggle?

**Comment by DJSdev** [comment]
I'm tried of every time I open Postman to do a quick test request, I get a fullscreen prompt 10-20 seconds after Postman is opened to asking to update. 

I don't want to update, I want to use the product, send 2 or 3 requests, and then close it. I don't care if I'm out of date, I'll update when I decide to.

Just found this issue and posting this to let the Postman team know I'm moving to a different tool.

**Comment by ZooDoo4U** [comment]
I found a great way to disable the auto update in postman.   Think most of us here have VS Code?   There is a great extension 'Thunder Client" and it does about 99.999% of what i need postman for...

**Comment by Stevemoretz** [comment]
Lol people are going on a strike here, for some just not using the Postman seems to have become a solution itself!

I do appreciate the OTA updates, and frankly I have them in the apps I develop too and mostly I don't (need to) let users to control those updates but my apps don't have offline support and my OTA updates are really small (a few KBs).

On the other hand Postman has huge updates and offline support, makes sense to have the option here, you can let users know that there is a critical security update and let them choose to update or not if security is the only concern.

Anyways I came here only cause I got email notifications... but found the people leaving postman just cause of this issue, made me sad; so I needed to say something...

For those who are sadly leaving, there is actually a workaround listed above which might help you.

**Comment by sathomps** [comment]
I can't use the latest version of Postman due to my machine being locked down (corporate env) and the Update.exe check failing on startup.

Guess it's time to move to [Thunder Client](https://marketplace.visualstudio.com/items?itemName=rangav.vscode-thunder-client)

**Comment by cdcrchrisparker** [comment]
How awesome that all this time later and this still isn't resolved, and the DEVs are ignoring requests.  Just yesterday I had someone walking me through a setup, and I started Postman as he requested.  He then said, "okay click on..." to which I replied, "it isn't ready to run yet - it's checking for updates".  I had to repeat that about 3 times, because of the amount of time it takes.

Eventually the _there's a later version_ dialog appeared, and he asked me why I don't update.  Because *I CAN'T*.  My user account literally cannot run the update!

How awesome that adding something to the local hosts file is the only thing available.  So please feel free to pop up the useless dialog absolutely every single time I start Postman.

I think I'm going to say awesome one more time.  Great work.

**Comment by arlemi** [comment]
@cdcrchrisparker Have you tried using Postman in your browser? It looks like it would solve most of the issues you're facing [go.postman.co](https://go.postman.co/)

**Comment by ZooDoo4U** [comment]
Using postman in the browser....  I do UI testing and need to check rest services along the way.  To avoid test failures first things my tests do is close any and all instances of any browser,  would give a worse user experience than not being able to stop the updates...  Thunder client in vs code has worked the best...

**Comment by ilias-sp** [comment]
Using postman from browser is like using a different product, and network conditions may prevent access to internet.

One potential solution to override this annoying issue, is to identify which domain postman uses to fetch the updates, and force the OS to resolve it to a fake IP (127.0.0.1). I didnt try it as i am not using postman anymore, but if anyone is willing to he/she can give it a try.

**Comment by FikaCode** [comment]
Wow, three years, this is a joke！

**Comment by cdcrchrisparker** [comment]
As an organization, we tend to use Postman.  We send and receive exported Postman collections all the time.  No doubt the browser version would work, _but_, if I were to receive a collection and then not be able to make it work - what would be the first thing the sender would do?  They would start looking for differences between my machine and theirs.  What would be the first thing they would see?  Postman for browsers.

I appreciate the suggestion.  I really do.  But no thanks.

Sad thing is that I'm sure the Postman folks want to receive usage metrics from the program, which is a fair reason not to turn off checks for minor updates.  Well guess what - you don't get them from me because I've used the hosts file trick mentioned by [Stevemoretz](https://github.com/Stevemoretz) above.  I found it by way of Stack Overflow, which led me here.  My hosts file looks like this:

0.0.0.0         dl.pstmn.io
0.0.0.0         sync-v3.getpostman.com
0.0.0.0         getpostman.com
0.0.0.0         go.pstmn.io

**Just a thought:** Have a way to completely turn off _automatic_ updates entirely.  Like has been asked for 100 times.  Don't turn off _checks_ for updates.  If potential updates are found, and the local version has updates turned off, then you show a (presumably red colored) information bar which shows something along the lines of "You are on version a.b.c and we now have version x.y.z available".

Everybody wins in this scenario: You get to have rolling patches which *most* people will receive.  Users who absolutely cannot update (like me) do not get an interrupted flow and can update when their system admin has the opportunity.  Users who are waiting to make sure the version is stable (like the O.P.) can run the update at their leisure.  Postman does the responsible thing and provides updates while simultaneously receiving usage metrics.  Those who wish to remain totally anonymous can still use the hosts file trick.

**Comment by RAFALAMAO** [comment]
Hi, in LINUX you can try this:

On file "AutoUpdaterService.js"; path: "Postman/app/resources/app/services/AutoUpdaterService.js", in line 35 only add "return false", like image. 

![image](https://user-images.githubusercontent.com/86860721/207995232-16d7d911-0cee-4db1-9c74-b42bc05caf3d.png)

**Comment by sojimaxi** [comment]
> **Just a thought:** Have a way to completely turn off _automatic_ updates entirely. Like has been asked for 100 times. Don't turn off _checks_ for updates. If potential updates are found, and the local version has updates turned off, then you show a (presumably red colored) information bar which shows something along the lines of "You are on version a.b.c and we now have version x.y.z available".
> 
I am still using postman v7. I don't care about the updates anymore since I blocked it. I launch it to test what I want then close it.

**Comment by stanleyxu2005** [comment]
Dear team. 

Although I have **disabled auto-update** the major versions, but it seems Postman still downloads minor verions automatically **without any notice**. Is this really good? 

I'm also very confused about why the file size of each updates is so huge. It is roughly 500MB per updates. So if I've received 10 updates, it would have used 5GB disk. 

![image](https://user-images.githubusercontent.com/5162687/214839467-eba9f841-bdc3-4110-bad8-0c013d6ee8da.png)

Each updates contains
1. A tailor made Electron application -> Postman.exe ~130MB
2. The main Postman application -> resources/app.asar ~314MB (**most applications, e.g. VSCode, Slack will not have that big size, unless there are some huge resource data embedded. So what is inside this file?**)
3. A minimal git incudled -> resources/data/git ~15MB

Here is my feedback to this issue:
1. Is it possible to delete previous installed versions, after the latest version is installed.
2. Can the update package to be much much smaller? It could be a delta patch or app.asar update only. No need to download 500MB every time, especially Postman minor releases pace is quite high.

**Comment by Zerogoki00** [comment]
@stanleyxu2005 add this to hosts file
```
0.0.0.0         dl.pstmn.io
0.0.0.0         sync-v3.getpostman.com
0.0.0.0         getpostman.com
0.0.0.0         go.pstmn.io
```

works for me. They are deprecating local workspace so I will never update anymore

**Comment by FikaCode** [comment]
Wow, Many years，If you are looking for a simple tool then I recommend Thunder Client, a plugin of Visual Studio Code.

**Comment by ZooDoo4U** [comment]
@FikaCode  Absolutely agree with your solution, even better after filing a few issues and suggestions regarding Thunder Client, i believe the longest from filing and fix has usually been less than a week.  Then again, did reverse engineer Postman's exported json to help generate classes to create code to do most operations for post/puts, gets... Guess if postman was more responsive and helpful, never would have written that stuff...

**Comment by nikhilbadyal** [comment]
Absolutly needed, I have postman installed on my organization device and there I don't have sudo access. So whenever I open postman. It downloads the update, wastes my bandwidth and comes up a pop up to install updated asking my sudo password. (which I don't have).

Also I use brew, so I would like brew to maintain my updates and not this auto update thing. Also even after updating with Auto Updator. Brew is going to update it anyway. Wasting my bandwidth twice.

<img width="275" alt="image" src="https://github.com/postmanlabs/postman-app-support/assets/59223300/4c1d86ef-0ca3-4e21-96af-ef1f2bf04ca7">

**Comment by ppanchal-mobiz** [comment]
> I can't use the latest version of Postman due to my machine being locked down (corporate env) and the Update.exe check failing on startup.
> 
> Guess it's time to move to [Thunder Client](https://marketplace.visualstudio.com/items?itemName=rangav.vscode-thunder-client)

I have the same problem as @sathomps. The machine is locked down so I disabled the updates since it kept failing. Now I have annoying popup everytime I start postman. How do I get rid of this?
![image](https://github.com/postmanlabs/postman-app-support/assets/104440431/1c625ff8-be64-4217-9776-7f98028d2ef5)

**Comment by cdcrchrisparker** [comment]
Scroll up just a little bit, and you'll see a trick from RAFALAMAO who commented on Dec 15, 2022.

**Comment by vladimirzind** [comment]
> Hi, in LINUX you can try this:
> 
> On file "AutoUpdaterService.js"; path: "Postman/app/resources/app/services/AutoUpdaterService.js", in line 35 only add "return false", like image.
> 
> ![image](https://user-images.githubusercontent.com/86860721/207995232-16d7d911-0cee-4db1-9c74-b42bc05caf3d.png)

Thank you for this. 
For Windows everything is packed in C:\Users\user\AppData\Local\Postman\app-9.31.28\resources\app.asar 

So what worked for me is: 

1. extract app.asar using: "npx asar extract app.asar <destfolder>"
2. rename app.asar to app.asar.old for backup
3. go to unpacked app.asar <destfolder>/services/AutoUpdaterService.js and edit it according to answer above
4. pack everything in unpacked app.asar folder with "npx asar pack . app.asar" 
5. place this new app.asar back to C:\Users\user\AppData\Local\Postman\app-9.31.28\resources\ 
6. run Postman

**Comment by MangEs23** [comment]
> Hi, in LINUX you can try this:
> 
> On file "AutoUpdaterService.js"; path: "Postman/app/resources/app/services/AutoUpdaterService.js", in line 35 only add "return false", like image.
> 
> ![image](https://user-images.githubusercontent.com/86860721/207995232-16d7d911-0cee-4db1-9c74-b42bc05caf3d.png)

thanks for the tips @RAFALAMAO  i really needed this because postman is no longer have scratch pad and force me to update the app to latest version where there is no scratch pad anymore

**Comment by frystalizer** [comment]
Given that we set to disable update. But the app on macOS does automatically update to replace the old version if I restart the app. What a shame.

**Comment by edu8rdo** [comment]
Nothing has changed here since 2019 - so the link to make the request is useless, clearly you don't care to address this. Just say that on top.

**Comment by ZooDoo4U** [comment]
@edu8rdo Can't say nothing has happened.   As i use VS Code there is an excellent plug-in "Thunder Client"  i use that, only thing i might be missing out on is running test scripts. While i don't use them i think i've seen comment they are possible, i just use it for simple API requests.  On the other side wrote my one library to import postman and/or swagger collections and generate all my requests, too lazy :)   I also found Insomnia from some youtube tutorials got it to, and it seems to work great as well and to be consistent with following along with the tutorial, works great no BS.  As for postman.... Good riddance :)

**Comment by alexolog** [comment]
Adding my voice to the issue, 5 years later.

Forced autoupdates with no way to revert are **evil**, and user-hostile.  Minor version updates can and do contain bugs.

**Comment by kate-tester-de** [comment]
Adding another voice. My company has an update process that includes Postman on a regular basis. If I have to update before then, I have to get permission and while it doesn't take long, it is an interruption to my workflow. Because of that, I just want to turn off automatic updates. I still want the updates but not to be asked to update once a week, which is what it feels like. While I get that you want people to update, you also need to understand that this is an annoyance to those of us that have security aware employers. Please give us this option!

**Comment by arlemi** [comment]
@kate-tester-de For deployment and updates at scale, your company may be interested in Postman Enterprise as it provides the ability to set a specific Postman version for the entire fleet of company devices. More info here: https://learning.postman.com/docs/administration/enterprise/managing-enterprise-deployment/

**Comment by JZADC** [comment]
Exact the same issue for updating on our side for user without admin priviliges
so 1+ from my side

So i have some workaround because we are using jamf as mdm but i cant understand why this is not possible from postman side to correct this behaviour. I also fund a case from 2017 with exact this issue
https://github.com/postmanlabs/postman-app-support/issues/3829. Now 2025 and no solved

![Image](https://github.com/user-attachments/assets/819c4c3c-b169-4145-a89f-9026b7029a51)

**Comment by built2order** [comment]
Another Enterprise customer asking for this issue to be prioritised. I've previous raised support requests with Postman Support as continuous admin prompts are a significant impact to productivity of users without local admin access.

**Comment by ggg17226** [comment]
i need a way to fully disable auto update, because in china mainland, the request to postman server is so slow and sometimes fail.
when i open the app, the app will running in a blank page and cannot operation at all.

**Comment by arlemi** [comment]
@built2order If you're an Enterprise customer, your IT admin has the ability to provide you with the Postman Enterprise app that can be locked to a specific version. More information here: https://learning.postman.com/docs/administration/enterprise/managing-enterprise-deployment/

**Comment by built2order** [comment]
> @built2order If you're an Enterprise customer, your IT admin has the ability to provide you with the Postman Enterprise app that can be locked to a specific version. More information here: https://learning.postman.com/docs/administration/enterprise/managing-enterprise-deployment/

Thanks for your comment. 

Unfortunately the version available on the enterprise console is often 2 or more months behind the public release. 

I'm interested in making available recent releases on a monthly basis rather than pinned release with functionality that is over 2 months the old.

**Comment by puzzle9** [comment]
in linux

You downloaded it but don't have permission to install it
Wasted traffic for nothing


```

03:02:14 postman.desktop[29309]: 1747076534563 main info "UpdateHandler~app-update-events - Received event",{"name":"updateElectron","namespace":"appUpdate","data":{"channel":"stable","version":"11.43.2","downloadVersion":"11.45.0-r250512-1424","appId":"3ac0c8a2-94c8-41df-8556-6810e02c2667","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Postman/11.43.2 Electron/31.3.1 Safari/537.36","platform":"LINUX","arch":"64","downloadURL":"https://dl.pstmn.io/download/version/11.45.0-r250512-1424/linux_64","updateServerDomain":"https://dl.pstmn.io/","additionalParamsString":"installationId=3ac0c8a2-94c8-41df-8556-6810e02c2667&userId=770491&syncEnabled=true&teamPlan=sync-free-3-trial&teamId=861146","userId":"770491"}}

03:02:14 postman.desktop[29309]: 1747076534563 main info "LinuxUpdater~downloadUpdate - UpdateInfo received",{"channel":"stable","version":"11.43.2","downloadVersion":"11.45.0-r250512-1424","appId":"3ac0c8a2-94c8-41df-8556-6810e02c2667","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Postman/11.43.2 Electron/31.3.1 Safari/537.36","platform":"LINUX","arch":"64","downloadURL":"https://dl.pstmn.io/download/version/11.45.0-r250512-1424/linux_64","updateServerDomain":"https://dl.pstmn.io/","additionalParamsString":"installationId=3ac0c8a2-94c8-41df-8556-6810e02c2667&userId=770491&syncEnabled=true&teamPlan=sync-free-3-trial&teamId=861146","userId":"770491"}

03:02:15 postman.desktop[29309]: 1747076535151 main error "@postman/app-updater: error",{"errno":-13,"code":"EACCES","syscall":"open","path":"/opt/postman/Postman-1747076534563.tar.gz","name":"Error","message":"EACCES: permission denied, open '/opt/postman/Postman-1747076534563.tar.gz'","stack":"Error: EACCES: permission denied, open '/opt/postman/Postman-1747076534563.tar.gz'"},{"info":"{\"packageName\":\"Postman\",\"eID\":\"Postman-1747076534563\",\"downloadURL\":\"https://dl.pstmn.io/download/version/11.45.0-r250512-1424/linux_64\",\"downloadDirectory\":\"/opt/postman\",\"downloadInfo\":{\"status\":\"DOWNLOAD_ERROR\",\"error\":{\"errno\":-13,\"code\":\"EACCES\",\"syscall\":\"open\",\"path\":\"/opt/postman/Postman-1747076534563.tar.gz\"},\"downloadDirectory\":\"/opt/postman\"}}"}

03:02:15 postman.desktop[29309]: 1747076535152 main info "UpdateHandler~app-update-events - Received event",{"name":"electronAppUpdateError","namespace":"appUpdate","data":{"error":{"errno":-13,"code":"EACCES","syscall":"open","path":"/opt/postman/Postman-1747076534563.tar.gz","name":"Error","message":"EACCES: permission denied, open '/opt/postman/Postman-1747076534563.tar.gz'","stack":"Error: EACCES: permission denied, open '/opt/postman/Postman-1747076534563.tar.gz'"},"updateData":{"packageName":"Postman","eID":"Postman-1747076534563","downloadURL":"https://dl.pstmn.io/download/version/11.45.0-r250512-1424/linux_64","downloadDirectory":"/opt/postman","downloadInfo":{"status":"DOWNLOAD_ERROR","error":{"errno":-13,"code":"EACCES","syscall":"open","path":"/opt/postman/Postman-1747076534563.tar.gz"},"downloadDirectory":"/opt/postman"}}}}

03:02:15 postman.desktop[29309]: 1747076535154 main warn "NewRelicService~RecordCustomEvent: API request failed",{"code":"ECONNREFUSED","name":"AggregateError","message":"","stack":"AggregateError [ECONNREFUSED]: \n    at internalConnectMultiple (node:net:1117:18)\n    at afterConnectMultiple (node:net:1684:7)"}


```

**Comment by sandeepcmsm** [comment]
postman team not even recognising this issue. there are ton of articles and stackoverflow question regarding the same. are postman team incapable to this simple. popping up every time you open an app is so non sense.

**Comment by dsignr** [comment]
> postman team not even recognising this issue. there are ton of articles and stackoverflow question regarding the same. are postman team incapable to this simple. popping up every time you open an app is so non sense.

They care only about just upselling you the enterprise version as is evident from this thread. Time to look for alternatives. 

Edit: For anyone coming here from Google and finding it disappointing that a tool meant specifically for developers adamantly decides to irritate them instead of addressing a basic issue with hopes of upselling an enterprise version will be happy to know that there's an open source alternative.

https://insomnia.rest
https://github.com/Kong/insomnia

Also, more importantly, it allows you to disable auto-updates.

<img width="716" alt="Image" src="https://github.com/user-attachments/assets/27efcc68-baaa-4f8e-adcb-d635d8bb5075" />

Bye bye Postman👋

**Comment by evic2132** [comment]
Now they have removed the option to turn off Auto Update altogether
I don't want it to auto download anything without me allowing it.

**Comment by ZooDoo4U** [comment]
I found the auto updates annoying, so found many other tools like "Thunder Client" which is a VS Code extension that works well for my needs,  or Insomnia and imaging any number of other tools as a Postman replacement.   Even started wrote a utility the i can read in a Postmand collection, hit a rest/swagger end point and build up the request myself to invoke the API calls.  No reason to suffer with Postman... :)

**Comment by dsignr** [comment]
> I found the auto updates annoying, so found many other tools like "Thunder Client" which is a VS Code extension that works well for my needs, or Insomnia and imaging any number of other tools as a Postman replacement. Even started wrote a utility the i can read in a Postmand collection, hit a rest/swagger end point and build up the request myself to invoke the API calls. No reason to suffer with Postman... :)

Just a heads up - Insomnia already imports Postman without any extra work involved. Another option is you can export to `cURL` and import it elsewhere too as `cURL` is pretty much standard on most of these apps.

**Comment by phillip-lesui-zocdoc** [comment]
@postman-staff can we get a status update on this issue?

**Comment by softmantk** [comment]
Still dont understand why postman team is not giving us any update regarding the issue.  This is really an annoying UX issue.

**Comment by meszaros-lajos-gyorgy** [comment]
I've solved this temporarily by adding `127.0.0.1   dl.pstmn.io` to my hosts file.

**Comment by sandeep-midi** [comment]
such non sense team. either their developers dont know how to ship backward compatible. or they are totaly arrogant fools.

**Comment by jamesburr** [comment]
Another upvote for this feature, every time I open Postman it prompts for "An update is ready to install. Postman is trying to add a new helper tool. Enter an administrator's name and password to allow this.".

Except I can't as it's a corporate laptop with no administrator rights. Please allow us to disable this and auto updates.

**Comment by AvaJbuHtiG** [comment]
Adding my voice to the issue, 6 years later.

**Comment by HakanKoseoglu** [comment]
Madness. First, tell us CLEARLY what you are attempting to install. I'm not going to assign a 3rd party app admin rights just because you asked nicely. Helper tool? What helper tool, give specifics!

**Comment by isaac-jensen** [comment]
One day...

**Comment by alexanderkiel** [comment]
I find it also annoying to get constantly update popups from macOS almost every day, while only having Postman open in the background, not having used it since a week. This noice should end. Otherwise Postman is still a good tool.

**Comment by apitukh** [comment]
This is very and very annoying. "An update is ready to install. Postman is trying to add a new helper tool." almost every day

**Comment by rabih** [comment]
It has been nearly 7 years since this issue opened and the fact that it still is an issue is very disappointing. The workaround to just blacklist the urls in the host file in a product built by developers for developers is just crazy to me and very disappointing since we pay for so many enterprise licenses.

**Comment by arlemi** [comment]
@rabih Enterprise users can use the Enterprise application and set a specific version of the app for the entire org. Please check the docs here: https://learning.postman.com/latest-v-12/docs/administration/enterprise/managing-enterprise-deployment

**Comment by gremki** [comment]
Unbelievable! There was a problem in 12.10 and I could not use Postman for a week or two, until 12.11 was released... No option to rollback, to disable updates, no solution but wait.
This is ridiculous.

**Comment by fix-a-lot** [comment]
It's really frustrating that I have to reconfigure my firewall settings every time the patch version bumps up. I'd love to be able to update only occasionally, whenever I feel like it.
